### ISSUE TITLE
Cancelling request type

### ISSUE THREAD

**Issue by gmq** [gund/ng-http-interceptor#160]
The README mentions that returning false in an interceptor cancels the request but I'm getting the following typing error:
```
Argument of type '(data: any[], method: string) => false | any[]' is not assignable to parameter of type 'Interceptor<any[], any[] | Observable<any[]>>'.
  Type 'false | any[]' is not assignable to type 'any[] | Observable<any[]>'.
    Type 'false' is not assignable to type 'any[] | Observable<any[]>'.
```
with this piece of code:

```typescript
httpInterceptor.request().addInterceptor((data, method) => {
        if (!auth.email || !auth.token) return false;
        const headers: Headers = getHttpHeadersOrInit(data, method);
        headers.set('X-User-Email', auth.email);
        headers.set('X-User-Token', auth.token);
        return data;
});
```

Any ideas about what I'm doing wrong? Thanks!

**Comment by gund** [comment]
Hey nice observation!
According to [this line](https://github.com/gund/ng-http-interceptor/blob/master/src/http/http-interceptor.ts#L5) the return type of interceptor function is defined as `any[] | Observable<any[]>` so there is no boolean.

That is basically why it complains =)

I will push the fix shortly

**Comment by gmq** [comment]
Wow, that was fast. Thanks!

**Comment by gund** [comment]
Fix landed in [v3.1.2](https://github.com/gund/ng-http-interceptor/releases/tag/v3.1.2)
